backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/sleketon/BpkSkeleton.kt

```diff
@@ -0,0 +1,189 @@
+        BpkSkeletonSizeType.Large -> BpkHeadlineSkeletonHeightLg
+        BpkSkeletonSizeType.None -> 0.dp
+    }
+
+    return this.height(height)
+}
+
+@Composable
+private fun shimmerBackgroundColor() : Color {
+    return dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint02)
+}
+
+@Composable
+private fun shimmerPrimaryColor() : Color {
+    return dynamicColorOf(Color(0x00FFFFFF), Color(0x00000000))
+}
+
+@Composable
+private fun shimmerSecondaryColor() : Color {
+    return dynamicColorOf(Color(0x99FFFFFF), Color(0x33000000))
+}
+
+@Composable
+private fun shimmerAnimation(): State<Dp> {
+    val infiniteTransition = rememberInfiniteTransition()
+    return infiniteTransition.animateValue(initialValue = (-500).dp, targetValue = 500.dp, typeConverter = Dp.VectorConverter,
+        animationSpec = infiniteRepeatable(animation = tween(durationMillis = 1000, delayMillis = 200, easing = LinearEasing)))
+
+}
+@Composable
+private fun ShimmerBox (modifier: Modifier) {
+    return Box(modifier = modifier
+        .background(
+            brush = Brush.horizontalGradient(
+                listOf(
+                    shimmerPrimaryColor(),
+                    shimmerSecondaryColor(),
+                    shimmerPrimaryColor(),
+                )
+            )
+        )
+    )
+}
+
+@Composable
+fun BpkImageSkeleton (
+    modifier: Modifier = Modifier,
+    cornerType: BpkSkeletonCornerType = BpkSkeletonCornerType.Square
+) {
+    val cornerRadius = if (cornerType === BpkSkeletonCornerType.Rounded) BpkBorderRadius.Sm else 0.dp
+    Box(modifier = modifier
+        .background(
+            shimmerBackgroundColor(),
+            RoundedCornerShape(cornerRadius)
+        ))
+}
+
+@Composable
+fun BpkBodyTextSkeleton (
+    modifier: Modifier = Modifier,
+) {
+    Box(modifier = modifier.height(BpkSpacing.Xxl)) {
+        Column(verticalArrangement = Arrangement.spacedBy(BpkSpacing.Md)) {
+            Box(modifier = Modifier.height(BpkSpacing.Md).fillMaxWidth(0.85f)
+                .background(shimmerBackgroundColor(), RoundedCornerShape(BpkSkeletonBorderRadiusXXS)))
+            Box(modifier = Modifier.height(BpkSpacing.Md).fillMaxWidth()
+                .background(shimmerBackgroundColor(), RoundedCornerShape(BpkSkeletonBorderRadiusXXS)))
+            Box(modifier = Modifier.height(BpkSpacing.Md).fillMaxWidth(0.57f)
+                .background(shimmerBackgroundColor(), RoundedCornerShape(BpkSkeletonBorderRadiusXXS)))
+        }
+    }
+}
+
+@Composable
+fun BpkHeadlineSkeleton (
+    modifier: Modifier = Modifier,
+    skeletonSize: BpkSkeletonSizeType = BpkSkeletonSizeType.None
+) {
+    Box(modifier = modifier.enhanceHeadlineSize(skeletonSize)
+        .background(shimmerBackgroundColor(), RoundedCornerShape(BpkBorderRadius.Sm)))
+}
+
+@Composable
+fun BpkCircleSkeleton (
+    circleDiameter:  Dp = BpkSpacing.Xl,
+    circleSize: BpkCircleSizeType = BpkCircleSizeType.None
```

Comment:
Here what you could do is turn the `BpkCircleSizeType` into a sealed class, rather than an enum and then have a custom type (rather than `None`) where you also include the diameter. This way there's no confusion between when the size vs diameter will be applied - something like this:

```
sealed class BpkCircleSizeType {
  object Small : BpkCircleSizeType()
  object Large : BpkCircleSizeType()
  class Custom(val diameter: Dp) : BpkCircleSizeType()
}
```

Reply:
Will update it.